lib/cassette/client/cache.rb

```diff
@@ -15,15 +16,16 @@ def initialize(logger)
       def fetch_tgt(options = {}, &_block)
         options = { expires_in: 4 * 3600, max_uses: 5000, force: false }.merge(options)
         fetch('Cassette::Client.tgt', options) do
-          self.clear_st_cache!
           logger.info 'TGT is not cached'
           yield
         end
       end
 
-      def fetch_st(service, options = {}, &_block)
+      def fetch_st(tgt, service, options = {}, &_block)
         options = { max_uses: 2000, expires_in: 252, force: false }.merge(options)
-        fetch("Cassette::Client.st(#{service})", options) do
+        hash = Digest::MD5.hexdigest(tgt)
```

Comment:
Do you think is really necessary to convert the TGT to MD5?

What problem do you think it can happen using the raw TGT in the cache key?

Doing this way it would be faster and it could help to debug the cache keys if necessary.

Reply:
But a simple key list would expose the actual value of the TGT

Reply:
this is not an issue on memcache but redis allows key listings and  the cache is used for file names when using the `tmp/` cache

Reply:
humm you're right!